src/puptoo/process/__init__.py

```diff
@@ -63,8 +60,10 @@ def extract(msg, extra, remove=True):
         except Exception as e:
             logger.exception("Failed to extract facts: %s", str(e), extra=extra)
             facts["error"] = str(e)
-        finally:
-            facts = postprocess(facts)
```

Comment:
postprocess is going to be called in the success and failure case.  postprocess looks for the "system_profile" key even though it might not exist and likely doesn't in the case of an error/exception.